extractor/src/main/java/org/schabi/newpipe/extractor/stream/Stream.java

```diff
@@ -1,108 +1,253 @@
+     * An integer to represent that the itag id returned is not available (only for YouTube, this
+     * should never happen) or not applicable (for other services than YouTube).
+     *
+     * <p>
+     * An itag should not have a negative value so {@code -1} is used for this constant.
+     * </p>
      */
-    @Deprecated
-    public final int format;
+    public static final int ITAG_NOT_AVAILABLE_OR_NOT_APPLICABLE = -1;
+
+    private final String id;
+    @Nullable private final MediaFormat mediaFormat;
+    private final String content;
+    private final boolean isUrl;
+    private final DeliveryMethod deliveryMethod;
+    @Nullable private final String baseUrl;
 
     /**
-     * Instantiates a new stream object.
+     * Instantiates a new {@code Stream} object.
      *
-     * @param url    the url
-     * @param format the format
+     * @param id             the ID which uniquely identifies the file, e.g. for YouTube this would
+     *                       be the itag
+     * @param content        the content or URL, depending on whether isUrl is true
+     * @param isUrl          whether content is the URL or the actual content of e.g. a DASH
+     *                       manifest
+     * @param format         the {@link MediaFormat}, which can be null
+     * @param deliveryMethod the delivery method of the stream
+     * @param baseUrl        the base URL of the content if the stream is a DASH or an HLS
+     *                       manifest, which can be null
      */
-    public Stream(String url, MediaFormat format) {
-        this(url, null, format);
+    public Stream(final String id,
+                  final String content,
+                  final boolean isUrl,
+                  @Nullable final MediaFormat format,
+                  final DeliveryMethod deliveryMethod,
+                  @Nullable final String baseUrl) {
+        this.id = id;
+        this.content = content;
+        this.isUrl = isUrl;
+        this.mediaFormat = format;
+        this.deliveryMethod = deliveryMethod;
+        this.baseUrl = baseUrl;
     }
 
     /**
-     * Instantiates a new stream object.
+     * Checks if the list already contains one stream with equals stats.
      *
-     * @param url        the url
-     * @param torrentUrl the url to torrent file, example https://webtorrent.io/torrents/big-buck-bunny.torrent
-     * @param format     the format
+     * @param stream the stream which will be compared to the streams in the stream list
+     * @param streamList the list of {@link Stream Streams} which will be compared
+     * @return whether the list already contains one stream with equals stats
      */
-    public Stream(String url, String torrentUrl, MediaFormat format) {
-        this.url = url;
-        this.torrentUrl = torrentUrl;
-        this.format = format.id;
-        this.mediaFormat = format;
+    public static boolean containSimilarStream(final Stream stream,
```

Comment:
> Shouldn't that be called _contains_

I don't think so, because we are not comparing the equality of streams, but only some of their properties (media format and delivery method). We also have two `equals` methods.

> I think it's more intuitive to have the streamList first

Probably, but this method may be used externally and is here since a while. We should take a look at this, but not in this PR, as the method can still be used externally and I am changing a lot of things in how `Stream`s work in the extractor.